packages/mdc-textfield/icon/_mixins.scss

```diff
@@ -42,14 +52,21 @@
 ///
 @mixin mdc-text-field-disabled-icon-color($color) {
   &.mdc-text-field--disabled {
-    @include mdc-text-field-icon-color_($color);
+    @include mdc-text-field-leading-icon-color_($color);
+    @include mdc-text-field-trailing-icon-color_($color);
   }
 }
 
 // Private mixins
 
-@mixin mdc-text-field-icon-horizontal-position_($position-property, $position, $padding, $input-padding) {
-  .mdc-text-field__icon {
+@mixin mdc-text-field-icon-horizontal-position_(
+  $icon-position,
+  $position-property,
+  $position,
+  $padding,
+  $input-padding
+) {
+  .mdc-text-field__#{$icon-position}-icon {
```

Comment:
Let's break the contents of this into two separate mixins: one for the icon positioning and one for the input positioning. We can drop the selectors inside each and then call them with the relevant selectors above.

For example, this mixin: https://github.com/material-components/material-components-web/blob/2167ee095b23315e90b4640f5edeb2c7936b438c/packages/mdc-textfield/_mixins.scss#L601-L613

would instead become

```
@mixin mdc-text-field-with-leading-icon_ {
  .mdc-text-field__icon--leading {
    @include mdc-text-field-icon-position-or-other-name_(...args go here);
  }

  .mdc-text-field__input {
    @include mdc-text-field-input-position-or-other-name_(...args go here);
  }

  .mdc-floating-label {
    @include mdc-rtl-reflexive-position(left, $mdc-text-field-icon-padding); 
  }
}
```

It's now much clearer what's being affected and we can make additional changes here without having to complicate the child mixins.

Reply:
I could do that work, but the end goal is that this mixin [will be completely removed](https://github.com/material-components/material-components-web/compare/feat/textfield/prefix-suffix#diff-f7c44d2746e4c925cd75a8940cae639dL49-L63) once we get rid of absolute positioning. Is it worth the effort to organize this mixin if we're just going to delete it shortly?

Reply:
Yeah, let's get it cleaned up. If for some reason we aren't able to get the `<label>` root element change in, we'll at least have improvements to the existing Sass.